### ISSUE TITLE
large gap of WER between python and runtime

### ISSUE THREAD

**Issue by wdbigbang** [wenet-e2e/wenet#612]
**Describe the bug**
We trained an model with the parameter "use_dynamic_left_chunk=true", while we want to use it on some online asr situation.
But when 40 epoch done, we find that there is large gap of wer% using recognize.py and runtime decode.sh。
Here is the exp results of our testsets.

num_left_chunk=8 WER% | num_left_chunk=-1 WER% |
-- | -- 
25 | 21.12
14.66 | 12.46
14.74 | 11.66
17.85 | 14.67
16.75 | 12.98

**Desktop (please complete the following information):**
 - OS: centos7
 - Version latest

**Comment by robin1001** [comment]
The gap of runtime and python? or the gap of num_left_chunk=8 and num_left_chunk=-1?

**Comment by wdbigbang** [comment]
Thank u for your reply.
It is the gap of runtime left_chunk=8 and python left_chunk=8.
And we've found out the reason is that there was a problem with our transformation of the runtime code, so the inputs of the encoder is not correct.
Since we corrected our mistakes, the difference between runtime and python immediately became acceptable

The latest results are shown below:

runtime num_left_chunk=8 WER% | python num_left_chunk=8 WER%
-- | --
21.31 | 21.12
12.67 | 12.46
14.74 | 11.66
14.72 | 14.67
13.30 | 12.98

Thank you again.
